src/python/pants/build_graph/target.py

```diff
@@ -473,7 +473,10 @@ def transitive_invalidation_hash(self, fingerprint_strategy=None, depth=0):
       def dep_hash_iter():
         for dep in self.dependencies:
```

Comment:
It looks like there is currently some disagreement in the definition of "strict" deps between https://github.com/pantsbuild/pants/blob/master/src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py#L66-L77 and this code.

The largest difference is that `CompileContext.strict_deps` expands aliases... a relatively minor bit is that it expands aliases.

Perhaps the FingerprintStrategy could actual be what computes the dep list?